tasks/argocd/src/main/java/com/walmartlabs/concord/plugins/argocd/ObjectMapper.java

```diff
@@ -72,4 +76,82 @@ public Map<String, Object> toMap(Object app) {
     public <T> T mapToModel(Map<String, Object> map, Class<T> model) throws IOException {
         return (T) readValue(writeValueAsString(map), model);
     }
+
+
+    public V1alpha1Application buildApplicationObject(TaskParams.CreateUpdateParams in) throws IOException {
+        Map<String, Object> metadata = new HashMap<>();
+        Map<String, Object> spec = new HashMap<>();
+        Map<String, Object> source = new HashMap<>();
+        Map<String, Object> helm = new HashMap<>();
+        Map<String, Object> destination = new HashMap<>();
+        Map<String, Object> body = new HashMap<>();
+
+        metadata.put("name", in.app());
+        metadata.put("namespace", ArgoCdConstants.ARGOCD_NAMESPACE);
+        metadata.put("finalizers", ArgoCdConstants.FINALIZERS);
+
+        if (in.annotations() != null) {
+            metadata.put("annotations", in.annotations());
+        }
+
+        destination.put("namespace", in.namespace());
+        destination.put("name", in.cluster());
+
+        if (in.gitRepo() != null) {
+            source.put("repoURL", Objects.requireNonNull(in.gitRepo()).repoUrl());
+            source.put("path", Objects.requireNonNull(in.gitRepo()).path());
+            source.put("targetRevision", Objects.requireNonNull(in.gitRepo()).targetRevision());
+        } else if (in.helmRepo() != null) {
+            source.put("repoUrl", Objects.requireNonNull(in.helmRepo()).repoUrl());
+            source.put("chart", Objects.requireNonNull(in.helmRepo()).chart());
+            source.put("targetRevision", Objects.requireNonNull(in.helmRepo()).targetRevision());
+        } else {
+            throw new RuntimeException("Source information not provided for " + in.app() + "." +
+                    "Provide either `gitRepo` or `helmRepo` details for the application to be created." +
+                    "Cannot proceed further. Refer docs (https://concord.walmartlabs.com/docs/plugins-v2/argocd.html#usage) for usage");
+        }
+
+        if (in.helm() != null) {
+            if (Objects.requireNonNull(in.helm()).parameters() != null)
+                helm.put("parameters", Objects.requireNonNull(in.helm()).parameters());
+
+            helm.put("values", Objects.requireNonNull(in.helm()).values());
+            source.put("helm", helm);
+        }
+        spec.put("project", in.project());
```

Comment:
`project` should be made mandatory variable incase of appset and should be passed as `in` parameters